lib/redcord/migration/index.rb

```diff
@@ -0,0 +1,24 @@
+# frozen_string_literal: true
+
+# typed: strict
+
+module Redcord::Migration::Index
+  extend T::Sig
+
+  sig { params(model: T.class_of(Redcord::Base), index_name: Symbol).void }
+  def remove_index(model, index_name)
+    if model.redis.sismember("#{model.model_key}:index_attrs", index_name)
+      attr_set_name = 'index_attrs'
+    elsif model.redis.sismember("#{model.model_key}:range_index_attrs", index_name)
+      attr_set_name = 'range_index_attrs'
+    else
+      raise(
+        Redcord::AttributeNotIndexed,
+        "#{index_name} is not an indexed attribute.",
+      )
+    end
+
+    model.redis.srem("#{model.model_key}:#{attr_set_name}", index_name)
+    model.redis.del("#{model.model_key}:#{index_name}")
```

Comment:
- For index attributes, wouldn't we want to scan through all values of the index attribute and delete it?
e.g. if `attr: a, String, index: true` is an attribute with values "a" and "b", then we'd have to delete the index sets `Redcord:Model:a:a` and `Redcord:Model:a:b` 

- I think this works for range index attributes; nil values for range index attributes are also specially stored as normal sets, so we would also want to delete `Redcord:Model:range_attr:`

Another potential concern is redis [del](https://redis.io/commands/del) time complexity is O(M) where M is the number of elements in the set. This could be potentially slow for the id set.

Reply:
> * For index attributes, wouldn't we want to scan through all values of the index attribute and delete it?
> nil values for range index attributes are also specially stored as normal sets, so we would also want to delete Redcord:Model:range_attr:

Ah, you're absolutely right! will  update and add a test case for these.

> Another potential concern is redis [del](https://redis.io/commands/del) time complexity is O(M) where M is the number of elements in the set. This could be potentially slow for the id set.

good concern! Though it might be a necessary operation?

Reply:
What do you think about using spop? https://redis.io/commands/spop

Reply:
I like it! feels like this could minimize the blocking caused by running migrations